pkg/operation/botanist/dns.go

```diff
@@ -66,7 +66,16 @@ func (b *Botanist) DeployExternalDNS(ctx context.Context) error {
 // the external DNSProvider and DNSEntry.
 func (b *Botanist) ExternalDNS() component.Deployer {
 	if b.NeedsExternalDNS() {
+		if b.isRestorePhase() {
```

Comment:
The code in this `if` is exactly the same as the one after it. Unless the order of deployment is changed (see other comments), this `if` is redundant and should be removed.